src/dmd/backend/oper.d

```diff
@@ -396,3 +388,269 @@ OPER swaprel(OPER);
 
 //#define EOP(e)  (!OTleaf((e)->Eoper))
 
+extern (D):
+
+extern (C) immutable ubyte[OPMAX] optab1 =
+() {
+    ubyte[OPMAX] tab;
+    foreach (i; Ebinary) { tab[i] |= _OTbinary; }
+    foreach (i; Eunary)  { tab[i] |= _OTunary;  }
+    foreach (i; Ecommut) { tab[i] |= _OTcommut; }
+    foreach (i; Eassoc)  { tab[i] |= _OTassoc;  }
+    foreach (i; Esideff) { tab[i] |= _OTsideff; }
+    foreach (i; Eeop0e)  { tab[i] |= _OTeop0e;  }
+    foreach (i; Eeop00)  { tab[i] |= _OTeop00;  }
+    foreach (i; Eeop1e)  { tab[i] |= _OTeop1e;  }
+    return tab;
+} ();
+
+immutable ubyte[OPMAX] optab2 =
+() {
+    ubyte[OPMAX] tab;
+    foreach (i; Elogical) { tab[i] |= _OTlogical; }
+    foreach (i; Ewid)     { tab[i] |= _OTwid;     }
+    foreach (i; Ecall)    { tab[i] |= _OTcall;    }
+    foreach (i; Ertol)    { tab[i] |= _OTrtol;    }
+    foreach (i; Eassign)  { tab[i] |= _OTassign;  }
+    foreach (i; Edef)     { tab[i] |= _OTdef;     }
+    foreach (i; Eae)      { tab[i] |= _OTae;      }
+    return tab;
+} ();
+
+immutable ubyte[OPMAX] optab3 =
+() {
+    ubyte[OPMAX] tab;
+    foreach (i; Eboolnop) { tab[i] |= _OTboolnop; }
+    return tab;
+} ();
+
+/*************************************
+ * Determine the cost of evaluating an operator.
+ *
+ * Used for reordering elem trees to minimize register usage.
+ */
+
+immutable ubyte[OPMAX] opcost =
+() {
+    ubyte[OPMAX] tab;
+    foreach (op; 0 .. OPMAX)
```

Comment:
This loop looks pretty inefficient, especially due to the inner loops. Rather use this?
```
foreach (o; Eunary) tab[o] += 2;
foreach (o; Ebinary) tab[o] += 7;
foreach (o; Elogical) tab[o] += 3;
tab[OPvar] += 1;
tab[OPmul] += 3;
...
```

Reply:
It only runs at compile time. The function doesn't even appear in the object file. Efficiency is irrelevant for it.

Reply:
Isn't compile time a thing, too? I just measured it: compiling oper.d with the loops takes about 100ms, that drops to 40ms with the proposed version. Not much, but to quote someone else: "these things add up"...

Reply:
I care about runtime performance of DMD, not so much about how long it takes to compile.

Also, as I explain endlessly, I tend to avoid refactoring code when I'm doing copy/pasting, for reasons I've repeated ad nauseum for 20 years.